pkg/edit/edit.go

```diff
@@ -29,7 +32,18 @@ import (
 // (for example due to a typo in the field name)
 // If there are extra fields it returns a string with a description of the diffs
 // If there are no extra fields it returns an empty string
-func HasExtraFields(yaml string, object runtime.Object) (string, error) {
+func HasExtraFields(yamlString string, object runtime.Object) (string, error) {
+	switch object.(type) {
+	case *kops.Cluster:
+		editedClusterObj := kops.Cluster{}
+		err := yaml.UnmarshalStrict([]byte(yamlString), &editedClusterObj)
+		if err == nil {
+			return "", nil
+		}
```

Comment:
Don't we also need to handle `kops.InstanceGroup`?

Reply:
We do.
There is very little testing around this. I think we should consolidate some of the logic in the edit commands with the replace command. Typos in yaml passed to replace is also a fairly common source of bugs.